### ISSUE TITLE
Regression in text field scalable text support

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#14583]
The `TextField` used to work for scaled text sizes, and now the label starts to overlap the text field at about 2.0, and continues to get worse.

![screenshot_20180208-182213](https://user-images.githubusercontent.com/8867023/36008795-d7bde81a-0cfd-11e8-83e6-312f01ab606c.png)

The developer should be able to put in pretty much any scale up to about 10.0 and get something expected (if not reasonable!).

**Comment by gspencergoog** [comment]
Tagging @HansMuller, because I think he was the last one to change it, but I'm happy to take a look at this too.

If you want my test app, also happy to supply that.
